src/client/Client.cc

```diff
@@ -9769,18 +9785,22 @@ int64_t Client::_preadv_pwritev_locked(Fh *fh, const struct iovec *iov,
         if (r <= 0)
           return r;
 
+        std::unique_lock cl{client_lock, std::adopt_lock};
```

Comment:
This isn't quite what I had in mind. It's weird to adopt the lock because if an exception were to occur, you'd unlock the mutex twice.

Again, the right thing to do here is pass the `std::unique_lock<>&` to the helper method.

Reply:
Sure, will fix it.